crates/cxx-qt-lib/src/core/qtlogging.rs

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+use cxx::{type_id, ExternType};
+use std::ffi::c_char;
+use std::ffi::CStr;
+use std::marker::PhantomData;
+
+#[cxx::bridge]
+mod ffi {
+    #[repr(i32)]
+    enum QtMsgType {
+        QtDebugMsg = 0,
+        QtInfoMsg = 4,
+        QtWarningMsg = 1,
+        QtFatalMsg = 3,
+        QtCriticalMsg = 2,
+        QtSystemMsg = 2
+    }
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        include!("cxx-qt-lib/qstring.h");
+        type QString = crate::QString;
+
+        include!("cxx-qt-lib/qtlogging.h");
+        type QMessageLogContext<'a> = crate::QMessageLogContext<'a>;
+        type QtMsgType;
+
+        fn qt_message_output(msgType: QtMsgType, context: &QMessageLogContext, string: &QString);
+
+        #[cxx_name = "qmessagelogcontext_line"]
+        fn line(context: &QMessageLogContext) -> i32;
+
+        #[cxx_name = "qmessagelogcontext_set_line"]
+        fn set_line(context: &mut QMessageLogContext, line: i32);
+
+        #[cxx_name = "qmessagelogcontext_file"]
+        unsafe fn file(context: &QMessageLogContext) -> *const c_char;
+
+        #[cxx_name = "qmessagelogcontext_set_file"]
+        unsafe fn set_file(context: &mut QMessageLogContext, file: *const c_char);
```

Comment:
I'm not a big fan of all the `setters` here.

For QMessageLogContext to be safe to use in Rust, it's lifetime needs to be bound to the lifetime of the `c_char*` it refers to.
The setters complicate this a lot, as they actually change the lifetime of the QMessageLogContext, which I don't even know how to express in Rust.
I'd recommend just providing a single constructor that takes `&'a CStr` for all the texts and binding QMessageLogContext to that lifetime.
For most use-cases it's probably fine if it's just immutable after construction, right?

Reply:
Yeah the changing-lifetime-thing is something I only discovered after doing the other PR :)

I also agree the setters are unnecessary, everything can be thrown into a constructor. Done